src/config.rs

```diff
@@ -37,96 +37,117 @@ pub static PUSH_REMOTE: &'static str = "push-remote";
 ///
 pub static PUSH_BRANCH: &'static str = "push-branch";
 
-fn load_from_file(path: &Path) -> io::Result<String> {
-    let mut file = try!(File::open(path));
-    let mut contents = String::new();
-    try!(file.read_to_string(&mut contents));
-    Ok(contents)
+pub struct Config {
+    toml: Table,
 }
 
-fn as_table(value: Value) -> Option<Table> {
-    match value {
-        Value::Table(s) => Some(s),
-        _ => None,
+impl Config {
+    fn load_from_file(path: &Path) -> io::Result<String> {
+        let mut file = try!(File::open(path));
+        let mut contents = String::new();
+        try!(file.read_to_string(&mut contents));
+        Ok(contents)
     }
-}
-
-///
-/// Parse the `Cargo.toml` file in the current directory and extract the keys
-/// under `package.metadata.sphinx`. This contains execution parameter defaults
-/// for the project using this cargo plugin.
-///
-pub fn parse_config() -> Result<Table, FatalError> {
-    let cargo_file_path = Path::new("Cargo.toml");
-    let cargo_file_content = try!(load_from_file(&cargo_file_path).map_err(FatalError::from));
 
-    let mut parser = Parser::new(&cargo_file_content);
-
-    let toml = parser.parse();
-    let package = toml.and_then(|mut table| table.remove("package").and_then(as_table));
-    let metadata = package.and_then(|mut table| table.remove("metadata").and_then(as_table));
-    let sphinx = metadata.and_then(|mut table| table.remove("sphinx").and_then(as_table));
+    fn as_table(value: Value) -> Option<Table> {
+        match value {
+            Value::Table(s) => Some(s),
+            _ => None,
+        }
+    }
 
-    sphinx.ok_or(FatalError::InvalidCargoFileFormat)
-}
+    ///
+    /// Parse the `Cargo.toml` file in the current directory and extract the keys
+    /// under `package.metadata.sphinx`. This contains execution parameter defaults
+    /// for the project using this cargo plugin.
+    ///
+    pub fn from(path: &str) -> Result<Config, FatalError> {
+        let path = Path::new(path);
+        let contents = try!(Config::load_from_file(&path).map_err(FatalError::from));
+
+        let mut parser = Parser::new(&contents);
+
+        let toml: Option<Table> = parser.parse();
+        let sphinx_toml = toml.and_then(|mut table| table.remove("package"))
+            .and_then(Config::as_table)
+            .and_then(|mut table| table.remove("metadata"))
+            .and_then(Config::as_table)
+            .and_then(|mut table| table.remove("sphinx"))
+            .and_then(Config::as_table);
+
+        let config: Table = try!(sphinx_toml.ok_or(FatalError::InvalidCargoFileFormat));
+
+        // Verify the TOML configuration.
+        let valid_keys = vec![DOCS_PATH, COMMIT_MESSAGE, SIGN_COMMIT, PUSH_REMOTE, PUSH_BRANCH];
+
+        for key in config.keys() {
+            if !valid_keys.contains(&key.as_ref()) {
+                println!("Unknown key \"{}\"", key);
```

Comment:
This is the only bit that wouldn't come nicely. Couldn't figure out the lifetimes for putting it in the `FatalError` response instead. :(